File: 2015/day-01/day-01-not-quite-lisp.js
const data = require("./day-01-data.json")

let floor = 0
let firstInstruction = Infinity
for(let i = 0; i < data.length; i++){
  if(data[i] === "(") floor++
  else floor--
  if(floor < 0) firstInstruction = Math.min(firstInstruction, i+1)
}

console.log(floor)
console.log(firstInstruction)

File: 2015/day-01/day-01-not-quite-lisp.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("./day-01-data.json")
        .expect("cannot read file");

    let mut floor = 0;
    let mut first_negative = 0;
    let mut index = 0;

    for c in input.chars() {
        match c {
            '(' => {
                floor += 1;
                index += 1;
            },
            ')' => {
                floor -= 1;
                index += 1;
                if floor < 0 && first_negative == 0 {
                    first_negative = index;
                }
            },
            _ => println!("End.")
        }
    };

    println!("Floor: {}", floor);
    println!("First negative: {}", first_negative);
}
